cedar-policy-core/src/parser/cst_to_ast.rs

```diff
@@ -5675,4 +5759,72 @@ mod tests {
                 .build(),
         );
     }
+
+    #[cfg(feature = "tolerant-ast")]
+    #[test]
+    fn multiple_policys_with_unparsable_policy_ok() {
+        // When we have a malformed policy, it should become an error node but the rest of the policies should parse
+        let policyset = text_to_cst::parse_policies_tolerant(
+            r#"
+            // POLICY 1 
+            @id("Photo.owner")
+            permit (
+            principal,
+            action in
+                [PhotoApp::Action::"viewPhoto",
+                PhotoApp::Action::"editPhoto",
+                PhotoApp::Action::"deletePhoto"],
+            resource in PhotoApp::Application::"PhotoApp"
+            )
+            when { resource.owner == principal };
+
+
+            // BEGIN ERROR POLICY NODE
```

Comment:
Was this intended for debugging?

Reply:
I meant it to clarify that the resulting AST would look like - I find it clearer, but can remove if it seems messy